app/assets/javascripts/utils/article_finder_utils.js

```diff
@@ -90,3 +90,30 @@ export const extractClassGrade = (pageAssessments) => {
   });
   return classGrade;
 };
+
+
+export const keywordAutocompleteGenerator = (keyword) => {
+  return {
+    list: 'search',
+    srsearch: keyword,
+    srlimit: 5,
+    srinfo: 'totalhits',
+    srprop: '',
+  };
+};
+
+
+const mediawikiApiBase = (language, project) => {
+  if (project === 'wikidata') {
+    return `https://${project}.org/w/api.php?action=query&format=json&origin=*`;
+  }
+  return `https://${language}.${project}.org/w/api.php?action=query&format=json&origin=*`;
+};
+
+export const fetchArticleAutocompleteResults = async (keyword, wiki) => {
+  const query = keywordAutocompleteGenerator(keyword);
+
+  return queryUrl(mediawikiApiBase(wiki.language, wiki.project), query).then((data) => {
+    return data.query.search.map(item => item.title);
+  });
+};
```

Comment:
Why not use `await` if the function is marked as `async`?

Reply:
Using await and return is redundant because then it will resolve the promise and again convert it into a promise because an async function returns a promise anyways.
See:
https://stackoverflow.com/a/44806250

Reply:
I meant in place of the `then`, the return is fine.

Reply:
If you meant the `then` I used on `queryUrl` then Yes I do have used await when I called this method but I have used `then` trying to abstract the API logic and its returned structure. If I remove it then every where I use it I need to know the underlying structure of the response JSON.